config/initializers/telephony.rb

```diff
@@ -11,19 +11,51 @@
   c.twilio.timeout = Figaro.env.twilio_timeout.to_i unless Figaro.env.twilio_timeout.nil?
   c.twilio.record_voice = Figaro.env.twilio_record_voice == 'true'
 
-  c.pinpoint.sms.region = Figaro.env.pinpoint_sms_region
-  c.pinpoint.sms.application_id = Figaro.env.pinpoint_sms_application_id
-  c.pinpoint.sms.shortcode = Figaro.env.pinpoint_sms_shortcode
-  c.pinpoint.sms.longcode_pool = JSON.parse(Figaro.env.pinpoint_sms_longcode_pool || '[]')
-  c.pinpoint.sms.credential_role_arn = Figaro.env.pinpoint_sms_credential_role_arn
-  if Figaro.env.pinpoint_sms_credential_role_arn.present?
-    c.pinpoint.sms.credential_role_session_name = Socket.gethostname
+  if Figaro.env.pinpoint_sms_configs.present?
+    JSON.parse(Figaro.env.pinpoint_sms_configs || '[]').each do |sms_json_config|
+      c.pinpont.add_sms_config do |sms|
+        sms.application_id = sms_json_config['application_id']
+        sms.region = sms_json_config['region']
+        sms.shortcode = sms_json_config['shortcode']
+        sms.longcode_pool = sms_json_config['longcode_pool'] || []
+        sms.credential_role_arn = sms_json_config['credential_role_arn']
+        if sms_json_config['credential_role_arn'].present?
+          sms.credential_role_session_name = Socket.gethostname
+        end
+      end
+    end
+  else
+    c.pinpont.add_sms_config do |sms|
+      sms.region = Figaro.env.pinpoint_sms_region
+      sms.application_id = Figaro.env.pinpoint_sms_application_id
```

Comment:
So this will accept the old-style code so if we forget to migrate an environment to the newer "everything" key things should still work